pkg/k8s/resource/resource.go

```diff
@@ -135,9 +143,34 @@ func (r *resource[T]) pushDelete(lastState any) {
 }
 
 func (r *resource[T]) Start(startCtx hive.HookContext) error {
+	r.wg.Add(1)
```

Comment:
Do we even really need Start() anymore?

Reply:
We still want this to wait until being told to start, e.g. we don't want "cilium-agent objects" inspection to actually start pulling data from api-server.